hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java

```diff
@@ -804,4 +806,21 @@ public static String regionNamesToString(final Collection<byte[]> regions) {
     sb.append("]");
     return sb.toString();
   }
+
+  /**
+   * WEB UI
+   */
+  public synchronized Map<RegionInfo, ServerName> getRegionLocations(TableName tableName) {
+    Map<RegionInfo, ServerName> tableRegions = new TreeMap<>();
+    for (RegionStateNode regionStateNode : regionsMap.tailMap(tableName.getName()).values()) {
+      if (!regionStateNode.getTable().equals(tableName)) {
+        break;
+      }
+      if (regionStateNode.toRegionState().isSplitOrMerge()) {
+        continue;
+      }
```

Comment:
We previously would see splitting/merged regions for a specific table, right? The RegionState should be listed in the table of Regions for this HBase table -- are not showing split/merge regions a big performance impact?